eng/build.ps1

```diff
@@ -62,6 +62,7 @@ param (
   [switch][Alias('test')]$testDesktop,
   [switch]$testCoreClr,
   [switch]$testCompilerOnly = $false,
+  [switch]$testCompilerEndToEndOnly = $false,
```

Comment:
Why do we need a separate switch? The test parallelization code has a granularity of `class`. That means a unit test assembly with a single `class` will always be run as a complete unit. Unsure why we need a separate runner here. 

Basically this switch is no different than running `dotnet test` in the EndToEnd directory. Not sure it's worth the cost to our build infrastructure.

Reply:
@jcouv didn't see an answer to this.

Reply:
I've removed the `testCompilerEndToEndOnly` from the public surface area of this script

Reply:
@jaredpar Reverted changes to `build.ps1`. Thanks